### PR TITLE
Swift 3.1 - rebuild all frameworks, clean up all swift warnings

### PR DESCRIPTION
**Pull request by bharat**
This PR resolves all swift 3.1 warnings.

### CODE REVIEW

**Comment by ps2** [comment]
Could you add the .travis.yml changes as well?

File: Loop/Managers/LoopDataManager.swift [hunk 1]
```diff
@@ -198,12 +198,13 @@ final class LoopDataManager {
 
         if insulinEffect == nil {
             updateGroup.enter()
-            deviceDataManager.doseStore.getGlucoseEffects(startDate: effectStartDate) { (effects, error) -> Void in
-                if let error = error {
+            deviceDataManager.doseStore.getGlucoseEffects(start: effectStartDate, end: .distantFuture) { result -> Void in
```
Comment (ps2) [inline Loop/Managers/LoopDataManager.swift]: The changes here, in StatusExtensionDataManager, PredictionTableViewController, and StatusTableViewController don't appear to be related to swift 3.1.  Could you put those into a different PR along with a description of what they're addressing?

Comment (bharat) [inline Loop/Managers/LoopDataManager.swift]: Swift 3.1 now throws warnings for using these deprecated APIs:
```
/Users/bharat/local/Loop/Loop/View Controllers/StatusTableViewController.swift:209:38: 'getRecentGlucoseValues(startDate:endDate:resultsHandler:)' is deprecated: Use getGlucoseValues(start:end:completionHandler:) instead
```

**Comment by bharat** [comment]
Since Swift 3.1 throws warnings for the deprecated APIs, I had rolled those up into this PR. But I can see the value in making those API changes separately so let me pull them out into a separate PR.
